### ISSUE TITLE
ModularTorchModel.fit_generator ignores the `loss` argument (dead code path)

### ISSUE THREAD

**Issue by ashishsoni-ai** [deepchem/deepchem#5074]
### Description

`ModularTorchModel.fit_generator` documents the `loss` parameter as:
> "a function of the form f(outputs, labels, weights) that computes the loss
> for each batch. If None (the default), the model's standard loss function is used."

However, the assigned `loss` variable is never actually used — `self.loss_func` is
called directly instead. This makes passing a custom `loss` to `fit()`/`fit_generator()`
silently a no-op.

There's already a FIXME acknowledging this in the code:

https://github.com/deepchem/deepchem/blob/master/deepchem/models/torch_models/modular.py#L84
https://github.com/deepchem/deepchem/blob/master/deepchem/models/torch_models/modular.py#L(fit_generator FIXME line)

```python
# FIXME This line is not needed as loss is computed inside the call to loss_func
if loss is None:
    loss = self._loss_fn
...
batch_loss = self.loss_func(inputs, labels, weights)   # `loss` var never used
```

### Reproduction

```python
"""
Repro: ModularTorchModel.fit_generator ignores the `loss` argument.
It always calls self.loss_func directly, never the `loss` variable
assigned from the `loss` parameter / self._loss_fn.
"""
import numpy as np
import torch
import deepchem as dc
from deepchem.models.torch_models.modular import ModularTorchModel

n_samples, n_feat, n_hidden, n_tasks = 6, 3, 2, 1
X = np.random.rand(n_samples, n_feat).astype(np.float32)
y = np.random.rand(n_samples, n_tasks).astype(np.float32)
dataset = dc.data.NumpyDataset(X, y)

components = {'linear': torch.nn.Linear(n_feat, n_hidden),
              'head': torch.nn.Linear(n_hidden, n_tasks)}
model = torch.nn.Sequential(components['linear'], components['head'])

modular_model = ModularTorchModel(model, components)

called_flags = {"custom_loss_called": False, "loss_func_called": False}

def custom_loss(outputs, labels, weights):
    called_flags["custom_loss_called"] = True
    raise RuntimeError("custom loss was called (unexpected if bug exists)")

def patched_loss_func(inputs, labels, weights):
    called_flags["loss_func_called"] = True
    return (torch.nn.functional.mse_loss(model(inputs), labels[0]) * weights[0]).mean()

modular_model.loss_func = patched_loss_func

# Pass a custom `loss` function explicitly - per docstring this SHOULD be used
modular_model.fit(dataset, nb_epoch=1, loss=custom_loss)

print("custom_loss_called:", called_flags["custom_loss_called"])
print("loss_func_called:", called_flags["loss_func_called"])

if not called_flags["custom_loss_called"] and called_flags["loss_func_called"]:
    print("BUG CONFIRMED: `loss` argument passed to fit() is silently ignored.")
else:
    print("No bug detected.")
```

Output:
custom_loss_called: False
loss_func_called: True
BUG CONFIRMED: loss argument passed to fit() is silently ignored.

### Impact

Any user relying on the documented `loss=` override in `fit()`/`fit_generator()` for a
`ModularTorchModel` subclass gets silently ignored behavior — no error, no warning.

### Proposed direction

Either:
1. Remove the dead `loss` parameter handling from `fit_generator` and update docs to
   clarify `loss` is not configurable for ModularTorchModel (loss is always defined via
   `loss_func`), or
2. Make `loss_func` respect an explicitly passed `loss` override.

Happy to open a PR once maintainers confirm which direction is preferred.
